app/controllers/application_controller.rb

```diff
@@ -1,4 +1,16 @@
 # frozen_string_literal: true
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  def append_info_to_payload(payload)
```

Comment:
Oh, and we would also need to do the same for the Admin & System app controllers, we could do it in a concern.